src/components/link/bl-link.ts

```diff
@@ -0,0 +1,166 @@
+import { CSSResultGroup, html, LitElement, TemplateResult } from "lit";
+import { customElement, property } from "lit/decorators.js";
+import { classMap } from "lit/directives/class-map.js";
+import { ifDefined } from "lit/directives/if-defined.js";
+import style from "./bl-link.css";
+
+export type LinkVariant = "inline" | "standalone";
+export type LinkSize = "large" | "medium" | "small";
+export type LinkKind = "primary" | "neutral";
+
+/**
+ * @tag bl-link
+ * @summary Baklava Link component for navigation
+ *
+ * @cssproperty [--bl-link-color=--bl-color-primary] Sets the color of link
+ * @cssproperty [--bl-link-hover-color=--bl-color-primary-hover] Sets the hover color of link
+ * @cssproperty [--bl-link-active-color=--bl-color-primary-active] Sets the active color of link
+ */
+
+@customElement("bl-link")
+export default class BlLink extends LitElement {
+  static get styles(): CSSResultGroup {
+    return [style];
+  }
+
+  /**
+   * Target URL for the link
+   */
+  @property({ type: String, reflect: true })
+  target = "";
+
+  /**
+   * Link variant - inline or standalone
+   */
+  @property({ type: String, reflect: true })
+  variant: LinkVariant = "inline";
+
+  /**
+   * Link size - only applies to standalone variant
+   */
+  @property({ type: String, reflect: true })
+  size: LinkSize = "medium";
+
+  /**
+   * Link kind - only applies to standalone variant
+   */
+  @property({ type: String, reflect: true })
+  kind: LinkKind = "primary";
+
+  /**
+   * Whether the link is external
+   */
+  @property({ type: Boolean, reflect: true })
+  external = false;
```

Comment:
do we need an `external` attribute here? adding an icon feels outside the boundaries of our design system, more like an UI kit addition

Reply:
It's nice to see an icon for external sites to show people they will leave I think, let's discuss